pkg/api/annotations.go

```diff
@@ -289,19 +289,59 @@ func (hs *HTTPServer) PatchAnnotation(c *models.ReqContext) response.Response {
 	return response.Success("Annotation patched")
 }
 
-func (hs *HTTPServer) DeleteAnnotations(c *models.ReqContext) response.Response {
-	cmd := dtos.DeleteAnnotationsCmd{}
-	if err := web.Bind(c.Req, &cmd); err != nil {
+func (hs *HTTPServer) MassDeleteAnnotations(c *models.ReqContext) response.Response {
+	cmd := dtos.MassDeleteAnnotationsCmd{}
+	err := web.Bind(c.Req, &cmd)
+	if err != nil {
+		return response.Error(http.StatusBadRequest, "bad request data", err)
+	}
+
+	if (cmd.DashboardId != 0 && cmd.PanelId == 0) || (cmd.PanelId != 0 && cmd.DashboardId == 0) {
```

Comment:
I am expecting to fail for that case. What I am trying to do here is to avoid inconsistent calls to this endpoint. If only panel or dash is provided then it would be always an invalid request, even if there is an `annotationId` along with them.